teachertool/src/components/ImportRubricModal.tsx

```diff
@@ -0,0 +1,99 @@
+import { useContext, useEffect, useState } from "react";
+import { AppStateContext } from "../state/appStateContext";
+import { Modal } from "react-common/components/controls/Modal";
+import { hideModal } from "../transforms/hideModal";
+// eslint-disable-next-line import/no-internal-modules
+import css from "./styling/ImportRubricModal.module.scss";
+import { getRubricFromFileAsync } from "../transforms/getRubricFromFileAsync";
+import { NoticeLabel } from "./NoticeLabel";
+import { Rubric } from "../types/rubric";
+import { RubricPreview } from "./RubricPreview";
+import { setRubric } from "../transforms/setRubric";
+
+export interface IProps {}
+
+export const ImportRubricModal: React.FC<IProps> = () => {
+    const { state: teacherTool } = useContext(AppStateContext);
+    const [selectedFile, setSelectedFile] = useState<File | undefined>(undefined);
+    const [selectedRubric, setSelectedRubric] = useState<Rubric | undefined>(undefined);
+    const [errorMessage, setErrorMessage] = useState<string | undefined>(undefined);
+
+    useEffect(() => {
+        async function updatePreview(file: File) {
+            const parsedRubric = await getRubricFromFileAsync(file, false /* allow partial */);
+            if (!parsedRubric) {
+                setErrorMessage(lf("Invalid rubric file."));
+            } else {
+                setErrorMessage(undefined);
+            }
+            setSelectedRubric(parsedRubric);
+        }
+
+        if (selectedFile) {
+            updatePreview(selectedFile);
+        } else {
+            setSelectedRubric(undefined);
+            setErrorMessage(undefined);
+        }
+    }, [selectedFile]);
+
+    function closeModal() {
+        setSelectedFile(undefined);
+        setErrorMessage(undefined);
+        setSelectedRubric(undefined);
+        hideModal("import-rubric");
```

Comment:
I wonder if hideModal shouldn't take a parameter. Just have it hide the current modal.